app/routes/user/endorsements.server.ts

```diff
@@ -67,14 +68,7 @@ export class EndorsementsServer {
       )
 
     const escapedEndorserString = endorserSub.replaceAll('"', '\\"')
-    const user = (
-      await client.send(
-        new ListUsersCommand({
-          UserPoolId: process.env.COGNITO_USER_POOL_ID,
-          Filter: `sub = "${escapedEndorserString}"`,
-        })
-      )
-    )?.Users?.[0]
+    const user = await this.#getCognitoUsernameForSub(escapedEndorserString)
     if (!user)
       throw new Response('Requested user does not exist', {
         status: 400,
```

Comment:
Move the exception into the `#getCognitoUsernameForSub` method, so that on success it always returns a valid user and never undefined?